### PR TITLE
xmlserializer: Fix initializer pattern

### PR DESCRIPTION
**Pull request by miguelgaio**
Move libxml2 destructor hook to XmlUtil.

Signed-off-by: Miguel Gaio <miguel.gaio@intel.com>
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-143259120%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23discussion_r40451788%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-143681752%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-143768949%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-143805303%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23discussion_r40641077%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-143982062%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-144084475%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-144085977%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/253%23issuecomment-143259120%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40dawagner%20%40krocard%20%40OznOg%20%40cc6565%20please%20review%22%2C%20%22created_at%22%3A%20%222015-09-25T15%3A50%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-28T08%3A48%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20%40OznOg%20%40cc6565%20please%20review%22%2C%20%22created_at%22%3A%20%222015-09-28T15%3A04%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%20T%20pull%20request%20message%20%28%60std%3A%3Arun_once%60%29%20is%20not%20correct%22%2C%20%22created_at%22%3A%20%222015-09-28T17%3A03%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-29T08%3A18%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-29T14%3A51%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22Please%20rebase.%22%2C%20%22created_at%22%3A%20%222015-09-29T14%3A57%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%208ee50801c3b7a1db442202681abdd85454504c65%20xmlserializer/XmlDocSource.cpp%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/253%23discussion_r40451788%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Could%20this%20also%20be%20used%20for%20the%20call%20to%20%60xmlInitParser%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-09-25T17%3A04%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-09-29T07%3A22%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xmlserializer/XmlDocSource.cpp%3AL52-59%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/253#issuecomment-143259120'>General Comment</a></b>
- <a href='https://github.com/miguelgaio'></a> @dawagner @krocard @OznOg @cc6565 please review
- <a href='https://github.com/dawagner'></a> :+1:
- <a href='https://github.com/miguelgaio'></a> @krocard @OznOg @cc6565 please review
- <a href='https://github.com/krocard'></a> :-1: T pull request message (`std::run_once`) is not correct
- <a href='https://github.com/miguelgaio'></a> :+1:
- <a href='https://github.com/OznOg'></a> :+1:
- <a href='https://github.com/dawagner'></a> Please rebase.
- [x] <a href='#crh-comment-Pull 8ee50801c3b7a1db442202681abdd85454504c65 xmlserializer/XmlDocSource.cpp 21'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/253#discussion_r40451788'>File: xmlserializer/XmlDocSource.cpp:L52-59</a></b>
- <a href='https://github.com/dawagner'></a> Could this also be used for the call to `xmlInitParser` ?

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/253?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/253?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/253'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by miguelgaio** [comment]
@dawagner @krocard @OznOg @cc6565 please review

File: xmlserializer/XmlDocSource.cpp [hunk 1]
```diff
@@ -50,7 +52,7 @@ CXmlDocSource::CXmlDocSource(_xmlDoc *pDoc, bool bValidateWithSchema,
       _strNameAttributeName(""),
       _bValidateWithSchema(bValidateWithSchema)
 {
-    init();
+    std::call_once(_initOnce, [](){ atexit(xmlCleanupParser); });
```
Comment (dawagner) [inline xmlserializer/XmlDocSource.cpp]: Could this also be used for the call to `xmlInitParser` ?

**Comment by dawagner** [comment]
:+1:

**Comment by miguelgaio** [comment]
@krocard @OznOg @cc6565 please review

**Comment by krocard** [comment]
:-1: T pull request message (`std::run_once`) is not correct

File: xmlserializer/XmlDocSource.cpp (hunk 1 above)
Comment (miguelgaio) [inline xmlserializer/XmlDocSource.cpp]: :+1:

**Comment by miguelgaio** [comment]
:+1:

**Comment by OznOg** [comment]
:+1:

**Comment by dawagner** [comment]
Please rebase.

**Pull request merged**